Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2985,6 +2990,9 @@ private void setActivityLifeCycleObserver(Application application) {
 
         @Override
         public void onActivityCreated(Activity activity, Bundle bundle) {
+            if(PromoViewHandler.getInstance().isInstallOrOpenPromoPending()){
+                PromoViewHandler.getInstance().showPendingPromoView(activity);
```

Comment:
We cannot fix this to  first or main activity. In most cases  app get initialize din splash and splash has a limited life time. So we are providing option for skipping promo views on any activity by implementing interface `IPromoViewControl`